### PR TITLE
Improve rustdocs for *_hash_ty methods

### PR DESCRIPTION
**Pull request by tcharding**
Recently we added two new methods with rustdocs. The rustdocs have a few minor things that can be improved.

- Use 'Returns' section for both success and error path returns
- Use markdown heading for section
- [Subjectively] improve wording of docs

@rish-singhal this PR is minor rustdocs stuff that I have recently been learning myself. I hope this makes your future patches easier to do in regards to the docs. FTR there are `# Returns` sections and `# Examples` sections that one can use, no set rules as far as I can tell. I chose `# Returns` in this case because it seemed to work. Please feel free to NACK the wording changes if you do not agree.

### CODE REVIEW

File: src/util/psbt/map/input.rs [hunk 1]
```diff
@@ -192,21 +192,23 @@ impl PsbtSigHashType {
 
 impl Input {
     /// Obtains the [`EcdsaSigHashType`] for this input if one is specified.
-    /// If no sighash type is specified, returns ['EcdsaSigHashType::All']
```
Comment (Kixunil) [inline src/util/psbt/map/input.rs:195]: IMO it was useful to have this as summary but a shorter wording may be nice

Comment (tcharding) [reply src/util/psbt/map/input.rs:195]: Cool, thanks, will re-spin after @rish-singhal has had a chance to comment.

**Review by apoelstra — APPROVED** [review]
ACK 3c74235ee18b33259b54b1d5bd17661d6b16c0b9

I don't understand the comment about the wording -- the old and new text look pretty-much identical to me.

**Comment by Kixunil** [comment]
@apoelstra there's no empty line so the difference is what goes to summary. I think that mentioning `All` in summary is useful.

**Comment by tcharding** [comment]
> I don't understand the comment about the wording -- the old and new text look pretty-much identical to me.

I was being particularly careful so as to make @rish-singhal feel welcome to the project.

**Review by apoelstra — APPROVED** [review]
ACK 7638d59fa66fc2bc5c27f956de4a20fa3e3558ba

**Review by Kixunil — APPROVED** [review]
ACK 7638d59fa66fc2bc5c27f956de4a20fa3e3558ba
